packages/agents/sequencer/src/lib/operations/auctions.ts

```diff
@@ -53,11 +54,30 @@ export const storeBid = async (
     });
   }
 
-  // TODO: Verify the following (and cache it for records):
-  // - Router is approved.
-  // - Asset is approved.
-  // - Relayer is approved for this chain (?).
-  // - Router has sufficient funds (?). May want to actually check this later.
+  // Sanity: Check router is approved.
+  const { router } = bid;
+  const { destinationDomain } = bidData.params;
+  let routerApproved = await cache.routers.getApproval(router, destinationDomain);
+  if (!routerApproved) {
+    // Regardless of whether the cache returned false or undefined, we want to check via subgraph
+    // now to see if router is approved.
+    routerApproved = await subgraph.isRouterApproved(destinationDomain, router);
```

Comment:
A router has to be approved to add liquidity so technically checking liquidity should cover all the cases of router approved, asset approved, and liquidity added.

Reply:
I mean... router could have been unapproved loll, but okay, that's cool with me

Reply:
Actually, I am sticking by this *for now*. It seems like it might be a useful/informative sanity check, esp. for new routers, if nothing else.

Left a TODO to later re-evaluate and remove if not useful. But for right now, it's clear we're struggling to have visibility on what's going wrong on sequencer.

Our hh tasks all check for it. For instance:
https://discord.com/channels/454734546869551114/964579852458012742/966795745988591627

Also: updated the expiry time for cached approval to be 24 hours

Reply:
I don't get it. The only place the approved routers are checked is when you're adding liquidity: https://github.com/connext/nxtp/blob/amarok/packages/deployments/contracts/contracts/Connext.sol#L644

So having liquidity necessarily means your router is approved. Why do we need to specifically check for approval?

Reply:
alright, i'll remove it